### DISCUSSION TITLE
Trim string Fields

### DISCUSSION THREAD

**Discussion by schmitch** [riok/mapperly#826] [Q&A] [answered]
is there any way to tell mapperly to auto trim input strings?

**Comment by TimothyMakkison** [comment, marked as answer]
Mapperly will use methods provided to it [see here](https://mapperly.riok.app/docs/configuration/static-mappers/)

```C#
[Mapper]
Public static class MyMapper
{
    // your mappers go here

    // mapperly will use this for all string -> string mapping
    Public static string Trimmer(string input) => input.Trim();
}
```

**Comment by randellhodges** [comment]
I haven't started using Mapperly yet and I'm in the research stage. I was wondering, can this be configured on a global/default level? Maybe something with MapperDefaults?

Basically, every time I map from string to string, anywhere, if the source is null or whitespace, I want to return null, otherwise I want to return the trimmed string. (I know, I know, feels hacky, but it is what it is)

In automapper, I do a ValueTransfers.Add<string>(...) once when setting up my mapper configuration and I never deal with it again.

**Reply by latonz** [reply]
You can use a base class Mapper with a user implemented mapping from string to string which trims the strings.

**Reply by randellhodges** [reply]
Of course. I'm not sure why I didn't think of that. I guess I need to get in there and try it out. Thanks for pointing that out :)
